Makefile

```diff
@@ -4,7 +4,8 @@ help:	## Show all Makefile targets.
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[33m%-30s\033[0m %s\n", $$1, $$2}'
 
 format:	## Run code autoformatters (black).
-	black .
+	pre-commit install
+	pre-commit run black-jupyter
```

Comment:
Maybe we should also have this with a third line `pre-commit run black-docs`, that way we maintain compatibility with the old `make format` of running on the full repo

Reply:
Also, I think you need `--all-files` so it runs on the full files, instead of just local Git changes

Reply:
Agreed. On using `all-files`, thanks for that.

Re: maintain compatibility with old `make format`, don't we have that already? Specifically: the command `pre-commit run black-jupyter --all-files` should run
- `black-src` which runs black formatting on everything but `/docs`
- `black-docs` which runs black formatting with `--line-length=79` arg

In other words, original `make format` runs black on full repo, and the above effectively does the same, or am I mistaken?

Reply:
Ah I see what you mean per id vs name.  You're right, I agree what you have here works.  Cheers!